### PR TITLE
Resolved issues related to column type changes

### LINKED ISSUE #1416: Trace file regeneration

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/blake2fmodexpdata/BlakeModexpDataOperation.java [hunk 1]
```diff
@@ -167,7 +167,7 @@ private void traceModexpResult(Trace trace, final UnsignedByte stamp) {
 
   private void commonTrace(Trace trace, UnsignedByte stamp, int index, Bytes input, int indexMax) {
     trace
-        .stamp(stamp)
+        .stamp(stamp.toInteger())
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/blake2fmodexpdata/BlakeModexpDataOperation.java:170]: Not sure this is the right way to do this.

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ext/ExtOperation.java
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/ext/ExtOperation.java:1]: ras

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:1]: ras

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/gas/GasOperation.java
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/gas/GasOperation.java:1]: ras

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/StackFragment.java [hunk 2]
```diff
@@ -185,14 +185,14 @@ public Trace trace(Trace trace) {
             trace::pStackStackItemPop3,
             trace::pStackStackItemPop4);
 
-    final List<Function<Bytes, Trace>> heightTracers =
+    final List<Function<Short, Trace>> heightTracers =
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/StackFragment.java:188]: `Short` was suggested by Intellij

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/common/CommonFragment.java [hunk 3]
```diff
@@ -93,8 +93,14 @@ public Trace trace(Trace trace) {
...
-        .height(commonFragmentValues.height)
-        .heightNew(commonFragmentValues.heightNew)
+        .height(
+            commonFragmentValues.hubProcessingPhase == HubProcessingPhase.TX_EXEC
+                ? commonFragmentValues.height
+                : 0)
+        .heightNew(
+            commonFragmentValues.hubProcessingPhase == HubProcessingPhase.TX_EXEC
+                ? commonFragmentValues.heightNew
+                : 0)
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/common/CommonFragment.java]: `HEIGHT` and `HEIGHT_NEW` vanish outside of execution rows.

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/logdata/LogData.java
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/logdata/LogData.java:1]: ras

**Comment by OlivierBBB** [comment]
What I'm most worried about is the change of the constraints commit 🤔

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/blake2fmodexpdata/BlakeModexpDataOperation.java (hunk 1 above)
Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/blake2fmodexpdata/BlakeModexpDataOperation.java:170]: looking at `UnsignedByte.java` it seems ok to me. It's used to convert to BigInt to

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ext/ExtOperation.java [hunk 4]
```diff
@@ -273,13 +272,13 @@ void trace(Trace trace, int stamp) {
           .ofJ(this.overflowJ[i])
           .ofI(this.overflowI[i])
           .ofRes(this.overflowRes[i])
-          .ct(Bytes.of(i))
-          .inst(Bytes.of(this.opCode.byteValue()))
+          .ct(i)
+          .inst(UnsignedByte.of(this.opCode.byteValue()))
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/ext/ExtOperation.java:276]: you could use `opCode.unsignedByteValue()`

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/logdata/LogData.java [hunk 5]
```diff
@@ -135,7 +134,7 @@ public void traceLog(final Log log, final int absLogNum, final int absLogNumMax,
           .logsData(true)
           .sizeTotal(log.getData().size())
           .sizeAcc(index == indexMax ? log.getData().size() : 16L * (index + 1))
-          .sizeLimb(index == indexMax ? UnsignedByte.of(lastLimbSize) : UnsignedByte.of(16))
+          .sizeLimb(index == indexMax ? lastLimbSize : 16)
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/logdata/LogData.java:137]: use `LLARGE`

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/tables/instructionDecoder/InstructionDecoder.java [hunk 6]
```diff
@@ -111,7 +113,8 @@ public int lineCount() {
 
   @Override
   public List<ColumnHeader> columnsHeaders() {
-    return Trace.headers(this.lineCount());
+    return headers(this.lineCount());
+    // return Trace.headers(this.lineCount());
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/tables/instructionDecoder/InstructionDecoder.java:117]: why keeping the comment ?

**Review by letypequividelespoubelles — APPROVED** [review]
only minor comment

**Pull request merged**